internal/host/validator.go

```diff
@@ -1229,18 +1229,179 @@ func (v *validator) printDefaultRoute(c *validationContext, status ValidationSta
+			if err := json.Unmarshal([]byte(host.APIVipConnectivity), &response); err != nil {
+				v.log.WithError(err).Warnf("Invalid API connectivity response")
+				continue
+			}
+
+			if !response.IsSuccess {
+				continue
+			}
+
+			return true
+		}
+	}
+
+	return false
+}
+
+// shouldValidateDNSResolution is utilized by the various DNS validation
+// functions determine whether they should perform DNS validation or not. If
+// not, it also returns the validation status they should have, which is
+// "pending" in some circumstances and "success" in others.
+//
+// In general, managed networking clusters don't require the user to sort out
+// DNS in advance, so we disable DNS validations while installing or adding
+// workers to managed networking clusters.
+//
+// If our SNO dnsmaq hack is enabled (ENABLE_SINGLE_NODE_DNSMASQ), then day-1
+// SNO clusters are an exception to the rule above. Our dnsmasq hack is enabled
+// for OCP and disabled for OKD.
+func (v *validator) shouldValidateDNSResolution(cluster *common.Cluster) (bool, ValidationStatus) {
+	var hasManagedNetworking bool
+	if !common.IsImportedCluster(cluster) {
+		hasManagedNetworking = !swag.BoolValue(cluster.UserManagedNetworking)
+	} else {
+		// The value of cluster.UserManagedNetworking is only correct for
+		// non-imported clusters, as imported clusters always have it set to
+		// false, regardless of its true value.
+		// This is why we have to use the following methods to try and guess
+		// the actual value.
+		// If `cluster.Imported` is nil we assume it's not imported, in order
+		// to maintain the older behavior for older clusters that may be running
+		// older agents that don't include the necessary information inside the
+		// ignition.
+		if !v.canDetermineImportedClusterManagedNetworking(cluster) {
+			// Wait until one of the day-2 hosts gets an API connectivity
+			// response so we can determine whether the cluster has managed
+			// networking or not
+			return false, ValidationPending
+		}
+
+		hasManagedNetworking = v.importedClusterHasManagedNetworking(cluster)
+	}
+
+	if hasManagedNetworking {
+		// Clusters with managed networking never need DNS validations as they
+		// automatically take care of the required DNS configuration within the
+		// host
+		return false, ValidationSuccess
+	}
+
+	if common.IsDay2Cluster(cluster) {
+		// All day 2 clusters that don't have managed networking, regardless of
+		// SNO or not, need DNS validations, as day-2 workers cannot benefit
+		// for the SNO dnsmasq hack that is levereged by the day-1 cluster.
+		return true, ""
+	}
+
+	// This is a day-1, user-managed-networking cluster
+
 	networkCfg, err := network.NewConfig()
 	if err != nil {
-		return false
+		// This should never happen, so the values chosen here are arbitrary
+		// and have no effect on anything
+		return false, ValidationSuccess
+	}
+
+	if common.IsSingleNodeCluster(cluster) && networkCfg.EnableSingleNodeDnsmasq {
```

Comment:
We take care of the DNS on the SNO, what about the day2 worker? do we take care of it's DNS configuration as well (as if it is being added to a clsuter with managedNetworking)?

Reply:
>We take care of the DNS on the SNO,

correct

>what about the day2 worker?

Irrelevant at this point in the function. See "	// This is a day-1, user-managed-networking cluster" comment above that indicates anything below that comment deals with day-1 UMN clusters

>do we take care of it's DNS configuration as well (as if it is being added to a clsuter with managedNetworking)?

We definitely could hack DNS for them like we do in day-1 SNO, but we currently don't do that, so instead we annoy the user with validations to make the user do it manually

Reply:
Once we decide to hack DNS for day-2 UMN, we could drop those validations. But it's a bit more involved as we can no longer use simple MC as that would apply to all workers, which might not be desired, so we need a new pool... and it gets complicated